test/testdata/rbi/string.rb

```diff
@@ -19,3 +19,8 @@
 
 w = "aãeéèbc".chars
 T.assert_type!(w, T::Array[String])
+
+# encoding overloadings
+"foo".encode("encoding", "other_encoding", fallback: {})
+"foo".encode("encoding", fallback: {})
+"foo".encode(fallback: {})
```

Comment:
These tests were added to make sure that `guessOverload` is using keyword arguments as a final hash argument.